pkg/compliant/auth/authoriser_builder.go

```diff
@@ -88,6 +97,55 @@ func (b AuthoriserBuilder) WithJwtExpiration(jwtExpiration time.Duration) Author
 	return b
 }
 
+func (b AuthoriserBuilder) WithSSAsPresence(ssasPresence bool) AuthoriserBuilder {
+	b.ssasPresence = ssasPresence
+	return b
+}
+
+func (b *AuthoriserBuilder) popSSAs() {
+	b.ssa = (b.ssas)[0]
+	if len(b.ssas) > 1 {
+		b.ssas = (b.ssas)[1:]
+	} else {
+		b.ssas = []string{}
+	}
+}
+
+// UpdateSSA - update the main ssa of the AuthoriserBuilder by popping the first one from ssas
+func (b *AuthoriserBuilder) UpdateSSA() error {
+	if len(b.ssas) == 0 && b.ssasPresence { // return error if there is not enogh SSAs
```

Comment:
1. ssaPresence  - rename to ssaPresent or isSsaPresent after checking current convention
2. Looks like ssaPresence flag is alway required, rewrite to make it obvious - exit early
3. enogh - typo